ClangTidy warnings for `misc-unused-parameters`

```
/Users/jviotti/Projects/core/src/core/json/json_value.cc:61:30: error: parameter 'value' is unused [misc-unused-parameters,-warnings-as-errors]
   61 | JSON::JSON(const Char *const value) : current_type{Type::String} {
      |                              ^~~~~
      |                               /*value*/

```

--- a/cmake/common/targets/clang-tidy.config
+++ b/cmake/common/targets/clang-tidy.config
@@ -11,7 +11,7 @@ Checks: '-*,
           performance-*,
           portability-*,
           objc-*,
-          misc-*,-misc-no-recursion'
+          misc-*,-misc-no-recursion,-misc-unused-parameters'
 WarningsAsErrors: '*'
 HeaderFilterRegex: ''
 FormatStyle: none

